src/services/todo-service.ts

```diff
@@ -1,45 +1,86 @@
-import { ITodo } from "../data-models/ITodo";
+import { ITodo } from "../data-models/interfaces/ITodo";
+
 
 const todos: ITodo[] = [
     { id: "1", title: "Task 1", complete: false },
     { id: "2", title: "Task 2", complete: false },
     { id: "3", title: "Task 3", complete: false },
 ];
 
-export const TodoService = {
-    getTodos: (): ITodo[] => {
-        return [...todos];
+
+
+interface ITodoService {
+    getTodos(): Promise<ITodo[]>;
+    removeTodo(id: string): Promise<boolean>;
+    addTodo(payload: ITodo): Promise<ITodo>;
+    toggleTodo(id: string, completed: boolean): Promise<boolean>;
+}
+
+
+
+export const TodoService: ITodoService = {
+
+    getTodos: (): Promise<ITodo[]> => {
+
+        return new Promise((resolve, _) => {
+            resolve([...todos]);
+        });
     },
+
+
+    removeTodo: (id: string): Promise<boolean> => {
+
+        return new Promise((resolve, _) => {
+
+            if (!id)
+            {
+                resolve(false);
+                return;
+            }
     
-    removeTodo: (id: string) => {
-        if (!id)
-        {
-            return;
-        }
-
-        
-       const index = todos.findIndex((item) => item.id === id);
-       todos.splice(index, 1);
+            const index = todos.findIndex((item) => item.id === id);
+            const arr = todos.splice(index, 1);
+            resolve(arr.length > 0);
+        });
     },
 
-    addTodo: (payload: ITodo) => {
-        if (!payload)
-        {
-            return;
-        }
 
-        
-        payload.id = crypto.randomUUID();
-        todos.push(payload);
-    },
+    addTodo: (payload: ITodo): Promise<ITodo> => {
 
-    toggleTodo: (id: string, completed:boolean) => {
+        return new Promise((resolve, reject) => {
 
-        const index = todos.findIndex((item) => item.id === id);
-        const item = todos[index];
-        todos[index] = {
-            ...item,
-            complete: completed
-        };
+            if (!payload)
+            {
+                reject();
```

Comment:
let's stick to one convention, either go with `reject` or `resolve(false)`. both are being used to handle almost the same case for no apparent reason